### DISCUSSION TITLE
Option to record only failing snapshots

### DISCUSSION THREAD

**Discussion by Scvairy** [pointfreeco/swift-snapshot-testing#936] [Q&A] [answered]
Now, when I need to regenerate changed snapshots, I regenerate all of them and have a massive diff in VCS (because newly created snapshots have different metadata?)
I'd like to minimize unnessecary snapshots (equal with the previous reference) being commited

Can I do that?

**Comment by mbrandonw** [comment, marked as answer]
We suppose this already with [`Record.failed`](https://swiftpackageindex.com/pointfreeco/swift-snapshot-testing/main/documentation/snapshottesting/snapshottestingconfiguration/record-swift.struct/failed). Pass that to [`withSnapshotTesting`](https://swiftpackageindex.com/pointfreeco/swift-snapshot-testing/main/documentation/snapshottesting/withsnapshottesting(record:difftool:operation:)-2kuyr) to configure the record mode.
